### PR TITLE
perf: lazy load tRPC routes

### PR DESCRIPTION
**Pull request by Mythie**
## What does this PR do?

This is an experimental branch that aims to address the current cold start issue that cal.com is facing. To do so we've swapped session fetching within tRPC to use a new slim handler which doesn't require `AUTH_OPTIONS`.

Additionally we've made the entire tRPC router lazy to avoid loading hundreds of dependencies during each API call.

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

Carefully.

### CODE REVIEW

File: packages/trpc/server/trpc.ts [hunk 1]
```diff
@@ -92,15 +92,20 @@ async function getUserFromSession({ session }: { session: Maybe<Session> }) {
...
 });
 
 const perfMiddleware = t.middleware(async ({ path, type, next }) => {
   performance.mark("Start");
+  const start = performance.now();
   const result = await next();
+  const end = performance.now();
   performance.mark("End");
   performance.measure(`[${result.ok ? "OK" : "ERROR"}][$1] ${type} '${path}'`, "Start", "End");
+  console.log(`[${result.ok ? "OK" : "ERROR"}][${end - start}ms] ${type} '${path}'`);
```
Comment (zomars) [inline packages/trpc/server/trpc.ts:108]: Doesn't this create duplicate logs?

**Comment by roae** [comment]
@Mythie I got ~15s in getSchedule  and 9s on session endpoints on cold boot

imagen

**Comment by Mythie** [comment]
Ya'll so brave 😅

**Comment by keithwillcode** [comment]
Initial testing results:
- Seeing issues on the `/api/book/event` endpoint. It's returning a 400 when attempting to book a new event

**Comment by keithwillcode** [comment]
Seeing same result on app.cal.dev so unrelated to this PR.

**Comment by keithwillcode** [comment]
Tested:
- Creating/Rescheduling/Deleting bookings
- Updating availability
- Creating and edit teams
- Updating settings
- Going through the app store
- Running insights

Verified all tRPC calls succeeded.

**Review by keithwillcode — APPROVED** [review]
Thanks again for the huge contribution, @Mythie!

**Comment by Mythie** [comment]
> Thanks again for the huge contribution, @Mythie!

Thank you guys for running with it! I know it’s a pain to make all those extra updates and test since this touches everything.

**Pull request merged**